openquakeplatform/openquakeplatform/updatures/app.py

```diff
@@ -0,0 +1,952 @@
+#!/usr/bin/env python
+# Copyright (c) 2014-2015, GEM Foundation.
+#
+# This program is free software: you can redistribute it and/or modify
+# under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import json
+import os
+import sys
+import copy
+import types
+from collections import OrderedDict
+from django.core.management import call_command
+from openquakeplatform import updatures
+from openquakeplatform.updatures.classes import (BackInheritance, ModelRefs)
+from openquakeplatform.updatures.models_descr import MODELS_DESCR
+from openquakeplatform import updatures
+
+MODELS_ORDER = []
+
+def rebuild_order():
+    reloop = True
+    first_loop = True
+    models_descr_old = MODELS_DESCR.copy()
+    models_descr_rest = {}
+    models_order = []
+    while reloop:
+        for model, descr in models_descr_old.iteritems():
+            updatures.UPD_LOG.log(10, "rebuild_order: model %s" % model)
+            if first_loop:
+                if descr.refs == {} and not descr.inher:
+                    models_order.append(model)
+                else:
+                    models_descr_rest[model] = descr
+            else:
+                for _, ref in descr.refs.iteritems():
+                    if isinstance(ref.model, types.FunctionType):
+                        continue
+                    if ref.model not in models_order:
+                        models_descr_rest[model] = descr
+                        break
+                else:
+                    if (not descr.inher) or (descr.inher in models_order):
```

Comment:
You don't need parenthesis, here you can just write

`if not descr.inher or descr.inher in models_order:`